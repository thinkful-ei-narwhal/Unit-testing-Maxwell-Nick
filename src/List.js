import React from 'react';
import Card from './Card';
import STORE from './store';
import './listStyles.css'

export default function List(props) {
  const cardComponents = props.cards.map(cardId => {
    const card = STORE.allCards[cardId];
    return <Card key={card.id} title={card.title} content={card.content} />;
  });

  return (
    <div className='List'>
      <header>
        {props.header}
      </header>
      <div className='List-cards'>
        {cardComponents}
      </div>
    </div>
  );
}
