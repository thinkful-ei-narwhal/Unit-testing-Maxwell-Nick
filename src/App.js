import React from 'react';
import List from './List';
import './appStyles.css'

function App(props) {
  const listComponents = props.store.lists.map(list => <List key={list.id} header={list.header} cards={list.cardIds} />);
  return (
    <main className='App'>
      <header>
        <h1>
          Trelloyes!
        </h1>
      </header>
      <div className='App-list'>
        {listComponents}
      </div>
    </main>
  );
}

export default App;
