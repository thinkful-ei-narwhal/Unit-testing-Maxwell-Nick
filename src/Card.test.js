import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import STORE from './store';
import Card from './Card';
import List from './List';
import renderer from 'react-test-renderer';

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Card key={STORE.allCards.a.id} title={STORE.allCards.a.title} content={STORE.allCards.a.content} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});