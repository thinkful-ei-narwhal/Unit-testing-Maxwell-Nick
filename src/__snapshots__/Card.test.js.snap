// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Card"
>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;
